src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -347,35 +455,47 @@ public TElement LastOrDefault(Func<TElement, bool> predicate)
 
     internal sealed class OrderedEnumerable<TElement, TKey> : OrderedEnumerable<TElement>
     {
-        internal OrderedEnumerable<TElement> parent;
-        internal Func<TElement, TKey> keySelector;
-        internal IComparer<TKey> comparer;
-        internal bool descending;
+        internal OrderedEnumerable<TElement> _parent;
+        internal Func<TElement, TKey> _keySelector;
+        internal IComparer<TKey> _comparer;
+        internal bool _descending;
```

Comment:
All of these could be `readonly` except `_parent` (that too with a tiny bit of refactoring).